python/elastic_search_serialization.py

```diff
@@ -0,0 +1,505 @@
+                "char_filter": {
+                    "dot_to_whitespace": {
+                        "type": "pattern_replace",
+                        "pattern": "(\\D)\\.(\\D)",  # split words on dots
+                        "replacement": "$1 $2",
+                    },
+                    "underscore_to_whitespace": {
+                        "type": "pattern_replace",
+                        "pattern": "_",  # split all on underscore
+                        "replacement": " ",
+                    },
+                },
+                "filter": {
+                    "asciifolding_original": {
+                        "type": "asciifolding",
+                        "preserve_original": True,
+                    }
+                },
+                "analyzer": {
+                    # we define our custom analyzer
+                    "lk_analyzer": {
+                        "tokenizer": "standard",
+                        "char_filter": [
+                            "dot_to_whitespace",
+                            "underscore_to_whitespace",
+                        ],
+                        "filter": ["asciifolding_original", "lowercase", "stop"],
+                    }
+                },
+            },
+        }
+    },
+    "mappings": {
+        "dynamic_templates": [
+            {
+                # store a copy of labels/types in a non-analyzed way for filtering
+                "lk_type_has_raw": {  # for edges, the key is "lk_type_has_raw"
+                    "match": "lk_type",  # for edges, use "lk_type"
+                    "mapping": {
+                        "type": "text",
+                        "analyzer": "lk_analyzer",  # (default "lk_analyzer")
+                        "fields": {"raw": {"type": "keyword"}},
+                    },
+                }
+            },
+            {
+                "string": {
+                    "match": "*_lke_string",
+                    "mapping": {
+                        "type": "text",
+                        "analyzer": "lk_analyzer",  # default: "lk_analyzer"
+                    },
+                }
+            },
+            {"date": {"match": "*_lke_date", "mapping": {"type": "date"}}},
+            {"number": {"match": "*_lke_number", "mapping": {"type": "double"}}},
+            {"boolean": {"match": "*_lke_boolean", "mapping": {"type": "keyword"}}},
+        ]
+    },
+}
+
+
+def serialize_vertex(vertex: mgp.Vertex) -> Dict[str, Any]:
+    """
+    Serializes vertex to specified ElasticSearch schema.
+    Args:
+        vertex (mgp.Vertex): Reference to the vertex in Memgraph DB
+    Returns:
+        Dict[str, Any]: ElasticSearch object representation.
+    """
+    node: Dict[str, Any] = {}  # whole part
+    node[ACTION] = {INDEX: {ID: str(vertex.id)}}
+    source = serialize_properties(vertex.properties.items())
+    source[LK_CATEGORIES] = [label.name for label in vertex.labels]
+    node[SOURCE] = source
+    return node
```

Comment:
You have different namings for the same thing, consider using vertex or node, not both. And same suggestion as below may be applied.